### PR TITLE
Fixed game commands being directly executed from within network_update.

### PR DESCRIPTION
**Pull request by ZehM4tt**
Refactored the network code to use the game command queue to not directly execute game commands from within network_update. This should also fix false positives when calling scenario_rand within game command callbacks.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
This will fail on none networked builds. You will need to add a empty definition for that.

File: src/openrct2/game.c [hunk 1]
```diff
@@ -361,6 +361,11 @@ void game_logic_update()
     gInUpdateCode = true;
     ///////////////////////////
 
+	//////////////////////////////////////////////////////////////////////////
```
Comment (marijnvdwerf) [inline src/openrct2/game.c]: tab

**Comment by ZehM4tt** [comment]
Sorry I've missed the define for non-network builds.

**Comment by ZehM4tt** [comment]
I've already set it to 14 in another PR, so it should vanish anyway.

File: src/openrct2/game.c [hunk 2]
```diff
@@ -361,6 +361,11 @@ void game_logic_update()
     gInUpdateCode = true;
     ///////////////////////////
 
+    //////////////////////////////////////////////////////////////////////////
```
Comment (Gymnasiast) [inline src/openrct2/game.c]: You're spamming banners. There is no need for most (or perhaps any) of them.

Comment (ZehM4tt) [inline src/openrct2/game.c]: I did not add the first one, will remove mine.

**Comment by janisozaur** [comment]
Requires rebase on top of #5631

File: src/openrct2/network/network.cpp [hunk 3]
```diff
@@ -1330,54 +1330,8 @@ void Network::ProcessPacket(NetworkConnection& connection, NetworkPacket& packet
...
-            if (game_command_queue.begin()->tick < gCurrentTicks) {
-                // Having old command from a tick where we have not been active yet or malicious server,
-                // the command is useless so lets not keep it.
-                game_command_queue.erase(game_command_queue.begin());
-
-                log_warning("Discarding game command from tick behind current tick, CMD: %08X, CMD Tick: %08X, Current Tick: %08X\n",
-                            gc.esi,
-                            gc.tick,
-                            gCurrentTicks);
+=======
```
Comment (AaronVanGeffen) [inline src/openrct2/network/network.cpp]: Rebase error?

Comment (janisozaur) [inline src/openrct2/network/network.cpp]: Whole rebase went south.

**Comment by ZehM4tt** [comment]
Sorry things went really wrong with this PR, please see https://github.com/OpenRCT2/OpenRCT2/pull/5642.

**Pull request closed without merging**